### DISCUSSION TITLE
Why jank-lang/jank uses merge commits instead of squash merging?

### DISCUSSION THREAD

**Discussion by haruki7049** [jank-lang/jank#131] [Q&A] [answered]
I like the `Squash merging` better than `Merge commits`, but [jank-lang/jank](https://github.com/jank-lang/jank) has used `Merge commits`. I want to know why [jank-lang/jank](https://github.com/jank-lang/jank) use `Merge commits` instead `Squash merging`.

**Comment by jeaye** [comment, marked as answer]
:shrug: Squashing is more work and merging is the default with git and github. I don't care about the existing of merge commits and, especially for (mainly) one dev, bundling whole features into one commit isn't really beneficial. Maybe, at some point, jank will have a commit policy which includes squashing, but I think it's extra process right now without much for tangible benefits. There's definitely a laziness aspect to it. :slightly_smiling_face:

**Reply by haruki7049** [reply]
I created this discussion. The reason is that I wanted someone to say, "We use merge commits with a reason", and I want someone to create `/CONTRIBUTING.md` file. The `/CONTRIBUTING.md` file is important to control manner of contribution in jank-lang/jank.

**Reply by wodin** [reply]
Another possibility (which also has its downsides) is to rebase and then merge with the `--no-ff` option. This way, you still have the individual commits and can bisect more easily, but the history is much flatter and neater.
The downsides being basically that rebase can sometimes do the wrong thing. But then again, so can merge. :man_shrugging:
